cmd/admin/main.go

```diff
@@ -117,22 +101,43 @@ func main() {
 
 	subcommand, ok := subcommands[unparsedArgs[0]]
 	if !ok {
-		cmd.FailOnError(errors.New("no recognized subcommand name provided"), "")
+		flag.Usage()
+		os.Exit(1)
 	}
 
+	// Then parse the rest of the args according to the selected subcommand's
+	// flags, and allow the global flags to be placed after the subcommand name.
+	subflags := flag.NewFlagSet(unparsedArgs[0], flag.ExitOnError)
+	subcommand.Flags(subflags)
+	flag.VisitAll(func(f *flag.Flag) {
```

Comment:
I think the "allow global flags..." comment from above should be moved to right above this line. I stared at it for a while before understanding the nuance that this is visiting all _global_ flags and redeclaring them as part of the `subflags` flag set. E.g.:

// Copy global flags into subcommand flags so, e.g., -config can appear after the subcommand.

Also there's an interesting nuance that `f.Value` is an interface value that winds up containing a pointer, so the `subflags.Var` call below can declare a flag that points to the same value as the original global flag did. Probably too in-the-weeds to include a comment, but interesting!

Reply:
Yeah, that's a good point -- the flag shenanigans here are quite magical. Done!